src/libraries/Microsoft.PowerFx.Core/Functions/UserDefinedFunction.cs

```diff
@@ -50,9 +50,19 @@ public override bool IsServerDelegatable(CallNode callNode, TexlBinding binding)
         {
             Contracts.AssertValue(callNode);
             Contracts.AssertValue(binding);
-            Contracts.Assert(binding.GetInfo(callNode).Function is UserDefinedFunction udf && udf.Binding != null);
-
-            return base.IsServerDelegatable(callNode, binding) || IsDelegatable;
+            Contracts.Assert(binding.GetInfo(callNode).Function is UserDefinedFunction udf && udf.Binding != null);
+
+            if (binding.DelegationHintProvider?.TryGetWarning(callNode, this, out var warning) ?? false)
+            {
+                SuggestDelegationHint(callNode, binding);

```

Comment:
>SuggestDelegationHint(callNode, binding); [](http://example.com/codeflow?start=16&length=41)

Should return false here. Delegation warning means that you can't delegate the call.

Reply:
removed change here